deploy/helm/elastic-agent/templates/agent/k8s/_pod_template.yaml

```diff
@@ -40,28 +41,8 @@ template:
       {{- . | toYaml | nindent 6 }}
     {{- end }}
     volumes:
-      {{- $definedAgentStateVolume := false -}}
-      {{- with ($presetVal).extraVolumes }}
+      {{- with ($agentVolumes).volumes }}
```

Comment:
I really liked the guarding you did on the _pod_template.yaml for the eck 😆 could we do the same here?

Reply:
oh you are right! this is always there. Missed this part ty